File: javascript/claude/p13_nr_ver2.js
const id_sequence = [10, 20, 30, 40, 50];

function getValue(index) {
    if (!Number.isInteger(index) || index < 0 || index >= id_sequence.length) return -1;
    return id_sequence[index];
}

const readline = require("readline");
const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

rl.question("Enter an index: ", (input) => {
    const trimmed = input.trim();
    const parsed = Number(trimmed);
    if (trimmed === "" || !Number.isInteger(parsed)) {
        console.log("Invalid input. Result: -1");
    } else {
        console.log("Result:", getValue(parsed));
    }
    rl.close();
});

File: c/claude/p13_nr_ver2.c
#include <stdio.h>

int id_sequence[] = {10, 20, 30, 40, 50};
const int SEQ_LEN = 5;

int get_value(int index) {
    if (index < 0 || index >= SEQ_LEN) return -1;
    return id_sequence[index];
}

int main() {
    int index;
    char leftover;
    printf("Enter an index: ");
    if (scanf("%d%c", &index, &leftover) != 2 || leftover != '\n') {
        printf("Invalid input. Result: -1\n");
        return 1;
    }
    printf("Result: %d\n", get_value(index));
    return 0;
}